src/Features/Core/Portable/CodeFixes/ICodeFixService.cs

```diff
@@ -11,11 +14,11 @@ namespace Microsoft.CodeAnalysis.CodeFixes
 {
     internal interface ICodeFixService
     {
-        Task<ImmutableArray<CodeFixCollection>> GetFixesAsync(Document document, TextSpan textSpan, bool includeSuppressionFixes, bool isBlocking, CancellationToken cancellationToken);
+        Task<ImmutableArray<CodeFixCollection>> GetFixesAsync(Document document, TextSpan textSpan, bool includeSuppressionFixes, bool isBlocking, CancellationToken cancellationToken, Func<string, IDisposable>? addOperationScope = null);
```

Comment:
❔ Why optional?

Reply:
I believe this API is used by TS, so making it non-optional will break them.

Reply:
Eliminating the original signature already broke binary compatibility, so might as well make this fully required.

Reply:
I instead added a new overload to ensure TS are not broken by this change.

Reply:
why/how  does TS use this service itself?

Reply:
if we have a new overload, we can make the variable non-optional in that overload.  that will clean up things *a lot* in terms of need to do null checks and whatnot.